test/functional/master/test_api_version_requests.py

```diff
@@ -0,0 +1,25 @@
+from genty import genty, genty_dataset
+
+from test.framework.functional.base_functional_test_case import BaseFunctionalTestCase
+
+@genty
+class TestMasterAPIVersionRequests(BaseFunctionalTestCase):
+
+    def _build_accept_header_with_api_version(self, version: int):
+        header = 'Accept'
+        value = 'application/vnd.clusterrunner.v{}+json'.format(version)
+        return {header: value}
+
+    @genty_dataset(
+        no_accept_header=(None, 1),
```

Comment:
Since `v1` is in the header in this particular example we'd want to return the first version for backwards compatibility.

I adjust the `UrlBuilder` class and add more tests without having the `v1` in the URL in my other branch (which I'm gonna make a PR with after this merges and I rebase). The reason I changed the `UrlBuilder` in the next branch instead of this one is because the next branch is makes the `v1` in the URL optional -- right now in this PR we still need to have the `v1` in the URL